### PR TITLE
descheduler: add mock client for decoupling test

### PR DESCRIPTION
**Pull request by Garrybest**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
According to #1262, add a mock grpc client so the descheduler could run the test by calling mock client.

I have moved service proto files into a new directory so that the mock client will not be handled by `go-to-proto` generation.

**Which issue(s) this PR fixes**:
Fixes part of #1262

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
@huone1 Could you please help take a look?

**Comment by huone1** [comment]
> @huone1 Could you please help take a look?

OK， I will review it as soon as possible in this week

File: hack/tools/tools.go [hunk 1]
```diff
@@ -1,10 +1,12 @@
+//go:build tools
```
Comment (RainbowMango) [inline hack/tools/tools.go:1]: You added this or the `gofmt` added this?

PS: If you are using Go v1.17 which will automatically add this tag.

File: pkg/estimator/service/doc.go [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package service
+
+//go:generate protoc --gogo_opt=paths=source_relative -I . -I ../../.. -I ../../../vendor --gogo_out=plugins=grpc:. --gogo_opt=Mpkg/estimator/pb/generated.proto=github.com/karmada-io/karmada/pkg/estimator/pb service.proto
+//go:generate mockery --name=EstimatorClient --inpackage
```
Comment (RainbowMango) [inline pkg/estimator/service/doc.go:4]: We just generate the mock code for EstimatorClient, but have not used it yet, right?

**Review by RainbowMango — COMMENTED** [review]
/lgtm

File: pkg/estimator/service/doc.go (hunk 2 above)
Comment (Garrybest) [reply pkg/estimator/service/doc.go:4]: Right, not used yet until the descheduler is contributed.

**Pull request merged**